### PR TITLE
Ft artisan request response 130385627

### PR DESCRIPTION
**Pull request by morriskimani**
#[130385627] Artisans should be able to accept or reject service requests

#### What does this PR do?
This PR implements adds functionality that allows an artisan, upon receiving a notification that a task has been assigned to him, to accept or reject the task. Should an artisan accept a quote, then he should be able to give a quote(price) and submit that quote. 
This PR also adds functionality for marking service request notifications as expired.
Finally, this PR adds functionality for notifying a tasker when his task has been accepted by an artisan provided by workdey.

#### Screenshots (if appropriate)
- The service request
notification_request

- On clicking accept, a quote field appears
notification_request_enter_quote

- Task is marked as accepted when artisan submits quote
task_accepted_with_quote

- Task is marked as rejected when artisan rejects it
task_rejected

### CODE REVIEW

File: app/controllers/notifications_controller.rb [hunk 1]
```diff
@@ -30,12 +30,12 @@ def update
     if record.update_attributes(@notifiable_attr_to_update.symbolize_keys)
       if @reply_to_sender == true
         @notification.reply_to_sender(@message, @event_name)
-        @notification.update_as_read
+        @notification.update_as_read force_update: true
```
Comment (hmasila) [inline app/controllers/notifications_controller.rb]: Why force the update?

File: spec/features/share_contact_spec.rb [hunk 2]
```diff
@@ -37,32 +37,19 @@
...
-           click_on "view quote"
-           click_button "Accept"
-
-           expect(page).to have_content(artisan.firstname)
-           expect(page).to have_content(artisan.email)
-         end
-     end
+         expect(page).to have_content(artisan.firstname)
+         expect(page).to have_content(artisan.email)
+       end
```
Comment (hmasila) [inline spec/features/share_contact_spec.rb]: check indentation

File: spec/features/share_contact_spec.rb [hunk 3]
```diff
@@ -72,8 +59,15 @@ def share_contact(share = true)
...
   end
 
-  def share_quote(share = true)
-    visit my_tasks_path
-    click_on ""
+  def send_quote
+    quote = create(:quote, artisan_id: artisan.id)
+    create(
+      :notification,
+      receiver_id:tasker.id,
```
[the remark this answers is not in the document]
Comment (hmasila) [reply spec/features/share_contact_spec.rb]: Put the space after colon

File: spec/features/share_contact_spec.rb (hunk 2 above)
Comment (morriskimani) [reply spec/features/share_contact_spec.rb]: done

File: app/controllers/notifications_controller.rb (hunk 1 above)
Comment (morriskimani) [reply app/controllers/notifications_controller.rb]: I need that to solve a bug where a  notification was being marked as `read` even before the artisan takes action i.e. accepts or rejects a quote

**Pull request merged**